tests/fixtures/tree-sitter-lint-plugin-replace-foo-with/src/lib.rs

```diff
@@ -1,18 +1,85 @@
 use std::sync::Arc;
 
-use tree_sitter_lint::{rule, violation, Plugin, Rule};
+use tree_sitter_lint::{
+    better_any::tid, rule, tree_sitter_grep::RopeOrSlice, violation, FileRunContext,
+    FromFileRunContext, FromFileRunContextInstanceProviderFactory, Plugin, Rule,
+};
 
-pub fn instantiate() -> Plugin {
+// pub type ProvidedTypes<'a> = ();
+pub type ProvidedTypes<'a> = (Foo<'a>,);
+
+#[derive(Clone)]
+pub struct Foo<'a> {
+    #[allow(dead_code)]
+    text: &'a str,
+}
+
+impl<'a> FromFileRunContext<'a> for Foo<'a> {
+    fn from_file_run_context(
+        file_run_context: FileRunContext<'a, '_, impl FromFileRunContextInstanceProviderFactory>,
+    ) -> Self {
+        println!("instantiating Foo for {:#?}", file_run_context.path);
+        Self {
+            text: match &file_run_context.file_contents {
+                RopeOrSlice::Slice(file_contents) => {
+                    std::str::from_utf8(&file_contents[..4]).unwrap()
+                }
+                _ => unreachable!(),
+            },
+        }
+    }
+}
+
+tid! { impl<'a> TidAble<'a> for Foo<'a> }
```

Comment:
This is presumably boilerplate that any provided-type author would need to provide